app/models/manageiq/providers/vmware/infra_manager/inventory/parser/virtual_machine.rb

```diff
@@ -274,20 +274,55 @@ def parse_virtual_machine_networks(vm, props, hardware, guest_devices)
       hostname = summary_guest[:hostName]
       guest_ip = summary_guest[:ipAddress]
       if hostname || guest_ip
+        # Find the ipStack that has dnsConfig present
+        ip_stack = props.fetch_path(:guest, :ipStack).to_a.detect { |ip| ip.dnsConfig }
+
         props.fetch_path(:guest, :net).to_a.each do |net|
+          ip_config_by_ip_addr = net.ipConfig&.ipAddress&.index_by(&:ipAddress) || {}
+
           ipv4, ipv6 = net[:ipAddress].to_a.compact.collect(&:to_s).sort.partition { |ip| ip =~ /([0-9]{1,3}\.){3}[0-9]{1,3}/ }
           ipv4 << nil if ipv4.empty?
           ipaddresses = ipv4.zip_stretched(ipv6)
 
+          dns_config = net.dnsConfig || ip_stack&.dnsConfig
+          if dns_config
+            domain_name = dns_config.domainName
+            dns_servers = dns_config.ipAddress.join(",") unless dns_config.ipAddress.empty?
+          end
+
+          ip_route_config = ip_stack&.ipRouteConfig
+          routes = ip_route_config&.ipRoute
+          if routes
+            default_routes = routes.select { |route| route.prefixLength == 0 }
+            default_ipv4_route = default_routes.detect { |route| route.network == "0.0.0.0" }
+            default_ipv6_route = default_routes.detect { |route| route.network == "::" }
+
+            default_gateway = default_ipv4_route&.gateway&.ipAddress || default_ipv6_route&.gateway&.ipAddress
+          end
+
           guest_device = guest_devices.detect { |gd| gd.data[:address] == net[:macAddress] }
 
           ipaddresses.each do |ipaddress, ipv6address|
+            netmask = if ipaddress
+              ip_config = ip_config_by_ip_addr[ipaddress]
+              IPAddr.new("255.255.255.255").mask(ip_config.prefixLength).to_s if ip_config
```

Comment:
@Fryguy do you know any slicker way of converting a subnet prefix length to a human readable subnet?

Reply:
You can play games with the fact that 255.255.255.255 is the same as 0xFFFFFFFF and the prefix length is how many leading bits you want, so maybe some shifts in there, but overall it's going to look uglier than what you have here which is pretty readable.

Only other thing that might be nice is create an extension method, like IPAddr.prefix_mask(prefix_length)

Reply:
Like:

```ruby
len = 24
IPAddr.new("255.255.255.255").mask(24)
# => "255.255.255.0"
(0xFFFFFFFF >> (32 - len) << (32 - len)).to_s(16)
# => 0xffffff00
```

Reply:
Yeah I was hoping for something like `IPAddr.subnet(24) => "255.255.255.0"`